### ISSUE TITLE
A ConnectionRefusedError on a "truesock" results in failures to use registered URIs for matching connections

### ISSUE THREAD

**Issue by sirosen** [gabrielfalcao/HTTPretty#406]
I'm not sure exactly what this issue comes from, but I noticed that my testsuite for an application works with HTTPretty < 1.x and fails with 1.x .
My current best guess is that I've been broken by [this change in 1.x](https://github.com/gabrielfalcao/HTTPretty/commit/f0bb2bdc1f5e2771e0cb854f57014e7d0fcc6b71).

I have code which uses pymemcache for caching, but memcached does not run on our current CI server. I found that tests work and pass when `memcached` is running, but not when it isn't. (Even though we treat failures to connect to memcached as cache misses.)
The cause appears to be that once there is a `ConnectionRefusedError`, subsequent HTTP(S) calls are not being passed through HTTPretty's URI matching anymore.

Here's a fairly minimal reproduction of the issue as it appears in our testsuite:

```python
>>> import requests, httpretty, pymemcache
>>> httpretty.enable()
>>> httpretty.register_uri("GET", "https://example.org/_dummy", body="foo")
>>> print(requests.get("https://example.org/_dummy"))
<Response [200]>
>>> pymemcache.Client(("localhost", 11211), timeout=1).get("foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/ebs/home/sirosen/venv-test/lib/python3.6/site-packages/pymemcache/client/base.py", line 498, in get
    return self._fetch_cmd(b'get', [key], False).get(key, default)
  File "/mnt/ebs/home/sirosen/venv-test/lib/python3.6/site-packages/pymemcache/client/base.py", line 863, in _fetch_cmd
    self._connect()
  File "/mnt/ebs/home/sirosen/venv-test/lib/python3.6/site-packages/pymemcache/client/base.py", line 296, in _connect
    sock.connect(self.server)
  File "/mnt/ebs/home/sirosen/venv-test/lib/python3.6/site-packages/httpretty/core.py", line 452, in connect
    self.connect_truesock()
  File "/mnt/ebs/home/sirosen/venv-test/lib/python3.6/site-packages/httpretty/core.py", line 490, in connect_truesock
    sock.connect(self._address)
ConnectionRefusedError: [Errno 111] Connection refused
>>> print(requests.get("https://example.org/_dummy"))
<Response [404]>
```

Note that this works fine on HTTPretty v0.9.7 -- that last request returns a 200 as expected, from the registered data.
In case it matters, I'm running on python3.6 .

I'm guessing that something became stateful which should not be, and that as a result, HTTPretty is going into a mode where it uses real connections for everything.

**Comment by tirumalesh-yeligar-by** [comment]
I am also facing the same issue

**Comment by gabrielfalcao** [comment]
the fix will be available in the next release (1.0.5)
